kivy/config.py

```diff
@@ -911,7 +911,30 @@ def name(self, value):
             try:
                 _, section, name = key.split("_", 2)
             except ValueError:
-                raise
+                Logger.warning((
+                    "Environ `{}` invalid format, "
+                    "must be KCFG_section_name").format(key))
+                continue
+
+            # extract and check section
             section = section.lower()
+            if not Config.has_section(section):
+                Logger.warning(
+                    "Environ `{}`: unknown section `{}`".format(
+                        key, section))
+                continue
+
+            # extract and check the option name
             name = name.lower()
+            sections_to_check = (
```

Comment:
OK, two  more comments since I looked this over again.  :-D

Consider bringing that list of sections to the top of the file as a named constant!  Or, not.

Reply:
I don't think so, that's not something i wanted to export/expose, it's just a matter of style to prevent having a big "if" but that's it

Reply:
Also have a super minor remark, this `sections_to_check` should probably be a set rather than a list

Reply:
I know this is in, but can't resist.  :-D 

It's a tuple actually, and if you replaced it (given you had infinite free time :-D), you might want to replace it with a `frozenset` as it's immutable, _but_ for small collections, `set`/`frozenset` is a little less efficient than a `tuple` and somewhat larger.